docs/attributes-registry/process.md

```diff
@@ -31,7 +32,26 @@
 | `process.session_leader.pid` | int | The PID of the process's session leader. This is also the session ID (SID) of the process. | `14` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | `process.user.id` | int | The effective user ID (EUID) of the process. | `1001` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | `process.user.name` | string | The username of the effective user of the process. | `root` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
-| `process.vpid` | int | Virtual process identifier. [1] | `12` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `process.vpid` | int | Virtual process identifier. [2] | `12` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 
-**[1]:** The process ID within a PID namespace. This is not necessarily unique across all processes on the host but it is unique within the process namespace that the process exists within.
+**[1]:** A globally unique identifier of the process. This can be any format
+of unique indentifier.
+
+On Linux, where possible, it is RECOMMENDED that this is calculated
+MD5 hash of the string `<pid>_<starttime>_<bootid>`.
```

Comment:
Given that we plan to introduce an [attribute for the OS thread id](https://github.com/open-telemetry/semantic-conventions/issues/812), and that `process.creation.time` and `host.id` already exist, I wonder whether you think it would make sense to utilize those (existing and well-defined) attributes?

Using start time as defined here or `process.creation.time` would not make any difference, I also think using `host.id` instead of `boot.id` doesn't have any drawbacks.

The main benefit of using those existing values and concepts would be that they are well-defined across platforms, and re-using them would decrease complexity for implementors.

Reply:
Unfortunately, I don't think any of those attributes will really work here to create a globally unique ID.

In Linux ( and I think most other OSs) OS Thread ID will repeat like PID will; when the system reaches MAX_PID it will rollover and repeat, so it's not a unique identifier on the system.

`process.creation.time` is based on the wall clock, which could repeat, if the system clock were to drift forward, and then be reset backwards to correct it.

`host.id` could be duplicated, especially if data collector were getting data from multiple networks/user sites.

Those are all good suggestions though, I thought of a lot of the same, until I found their drawbacks.

I hope this is method is easy for implementers . I think most will already be tracking PID and start time, so the only thing they need to do is read the boot_id, which should just be a single read, once when the collector starts. I know reading it might not be possible in containers or because of other system restrictions, so that is a limitation of this method.